tests/vmi_configuration_test.go

```diff
@@ -346,6 +347,42 @@ var _ = Describe("Configurations", func() {
 				Expect(err).ToNot(HaveOccurred())
 			})
 		})
+
+		Context("with guestAgent", func() {
+			var agentVMI *v1.VirtualMachineInstance
+			BeforeEach(func() {
+				agentVMI = tests.NewRandomVMIWithEphemeralDisk(tests.RegistryDiskFor(tests.RegistryDiskAlpine))
+			})
+
+			It("there should be agent channel when enabled", func() {
+				agentVMI.Spec.GuestAgent = &v1.GuestAgent{}
+				By("Starting a VirtualMachineInstance")
+				agentVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(agentVMI)
+				Expect(err).ToNot(HaveOccurred(), "Should create VMI successfully")
+				tests.WaitForSuccessfulVMIStart(agentVMI)
+
+				getOptions := metav1.GetOptions{}
+				var freshVMI *v1.VirtualMachineInstance
+
+				freshVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(agentVMI.Name, &getOptions)
+				Expect(err).ToNot(HaveOccurred(), "Should get VMI ")
+
+				domain := &wrapapi.Domain{}
+				context := &wrapapi.ConverterContext{
+					VirtualMachine: freshVMI,
+					UseEmulation:   true,
+				}
+				wrapapi.Convert_v1_VirtualMachine_To_api_Domain(freshVMI, domain, context)
```

Comment:
This test isn't verifying that the channel is opened to the guest agent.

Instead you're retrieving an already created VMI, then doing a local domain conversion on the retrived VMI to verify the resulting domain spec has the channel defined. That's the same thing you do in the unit test. We don't need a functional test to verify the conversion.